libraries/AP_NavEKF3/AP_NavEKF3_Control.cpp

```diff
@@ -411,8 +411,7 @@ void NavEKF3_core::checkAttitudeAlignmentStatus()
     // Once the tilt variances have reduced to equivalent of 3deg uncertainty, re-set the yaw and magnetic field states
     // and declare the tilt alignment complete
     if (!tiltAlignComplete) {
-        Vector3f angleErrVarVec = calcRotVecVariances();
-        if ((angleErrVarVec.x + angleErrVarVec.y) < sq(0.05235f)) {
+        if (tiltErrorVariance < sq(radians(5.0f))) {
```

Comment:
Yes, it was taking too long for the tilt alignment to pass. I will update the comment. The tiltAlignCompleteFlag is initialised to false and calcTiltErrorVariance() is run every covariance prediction.

Reply:
I still think there is a danger of tiltAlignComplete being set when we haven't run calcTiltErrorVariance(). Even if it can't happen with the current code, the flow of control between controlFilterModes(), checkAttitudeAlignmentStatus() and calcTiltErrorVariance() is a bit too complex.
Some possible fixes:
 - init tiltErrorVariance to a large number in InitialiseVariables
 - call calcTiltErrorVariance() from within checkAttitudeAlignmentStatus() when tiltAlignComplete is false

Reply:
tiltErrorVariance has been initialised to a large number in InitialiseVariables()